dev/PushNotifications/PushNotificationsLongRunningTask/NotificationListener.cpp

```diff
@@ -22,10 +22,17 @@ STDMETHODIMP_(HRESULT __stdcall) NotificationListener::OnRawNotificationReceived
     if (!m_foregroundSinkManager->InvokeForegroundHandlers(m_appId, payloadArray, payloadLength))
     {
         std::string commandLine = "----WindowsAppSDKPushServer:-Payload:\"";
-        commandLine.append(reinterpret_cast<char*>(payload), payloadLength);
+
+        // Escape special characters to follow command line standards for any app activation type in AppLifecycle
+        // (See AppInstance.cpp and Serialize() from other activation types)
+        std::wstring payloadAsWideString = GetPayloadAsWideString(payloadLength, payload);
```

Comment:
Is there an API to directly add escape components to UTF-8 string? Right now we have UTF8->UTF16->UTF8 which seems to be highly redundant

Reply:
I'm thinking that the utf16 conversion is simply unecessary overhead here.

Reply:
There's no such API ☹️. The closest I found is [UrlEscapeA](https://docs.microsoft.com/en-us/windows/win32/api/shlwapi/nf-shlwapi-urlescapea) which seem to be able to do the same encoding as the winrt Uri does. The caveats are:

1. The input is not a URL.
2. It does the conversion up to a maximum of INTERNET_MAX_URL_LENGTH (2k characters, evidently insufficient).

What I can do is to use the ShellExecuteEx version for wide strings. This way I avoid the payload wstring->string conversion, as well as the process name wstring->string conversion too.

Reply:
Can you quickly elaborate on the total number of string conversions that we do for a Push scenario in Reunion today? 

At the end of the day, we give the developer a stream of bytes. So we want to avoid any unecessary wide character utf-16 conversions unless absolutely necessary.

Reply:
We have another constraint at AppLifecycle: the command line is a wide string. This is why we end up doing another conversion at PushNotificationReceivedEventArgs to convert it to a byte array. So, the big picture looks like this:

WNS -> wpncore -> NotificationListener : byte array
NotificationListener -> ShellExecuteEx : byte array -> wide string (+1 conversion)
**AppLifecycle -> GetRawNotificationEventArgs -> PushNotificationReceivedEventArgs : wide string**
GetRawNotificationEventArgs (BuildPayload) : wide string -> byte array (+1 conversion)